recipes/capnproto/all/conanfile.py

```diff
@@ -0,0 +1,148 @@
+    license = "MIT"
+    topics = ("conan", "capnproto", "serialization", "rpc")
+    homepage = "https://capnproto.org"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ("CMakeLists.txt", "patches/*")
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "lite": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "lite": False
+    }
+
+    _cmake = None
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "4.3",
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 14)
+        mininum_compiler_version = self._minimum_compilers_version.get(str(self.settings.compiler))
+        if mininum_compiler_version and tools.Version(self.settings.compiler.version) < mininum_compiler_version:
+            raise ConanInvalidConfiguration("Cap'n Proto doesn't support {0} {1}".format(self.settings.compiler, self.settings.compiler.version))
+        if self.settings.compiler == "Visual Studio" and self.options.shared:
+            raise ConanInvalidConfiguration("Cap'n Proto doesn't support shared libraries for Visual Studio")
+
+    def requirements(self):
+        if not self.options.lite:
+            self.requires("zlib/1.2.11")
+            self.requires("openssl/1.1.1g")
+
+    def build_requirements(self):
+        if self.settings.os == "Linux":
+            self.build_requires("autoconf/2.69")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_TESTING"] = False
+        self._cmake.definitions["EXTERNAL_CAPNP"] = False
+        self._cmake.definitions["CAPNP_LITE"] = self.options.lite
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        configure_dir=os.path.join(self._source_subfolder, "c++")
+        with tools.chdir(configure_dir):
+            self.run("{} --install --verbose -Wall".format(tools.get_env("AUTORECONF")))
```

Comment:
AUTORECONF is not set here, and it produces error:
```

----Running------
> None --install --verbose -Wall
-----------------
/bin/sh: None: command not found

```

Reply:
yes, autoconf was requested in build_requirements() on Linux only, but it's required for non-Windows actually.